src/pudl/transform/ferc1.py

```diff
@@ -7,1400 +7,763 @@
+        different units, convert based on typical values. E.g. 19.85 MMBTU per ton of
+        coal, 1.037 Mcf per MMBTU of natural gas, 7.46 barrels per ton of oil.
+
+          * solid fuels (coal and waste): short tons [ton]
+          * liquid fuels (oil): barrels [bbl]
+          * gaseous fuels (gas): thousands of cubic feet [mcf]
+
+        Columns to which these physical units apply:
+
+          * fuel_consumed_units (tons, bbl, mcf)
+          * fuel_cost_per_unit_burned (usd/ton, usd/bbl, usd/mcf)
+          * fuel_cost_per_unit_delivered (usd/ton, usd/bbl, usd/mcf)
+
+        One remaining challenge in this standardization is that nuclear fuel is reported
+        in both mass of Uranium and fuel heat content, and it's unclear if there's any
+        reasonable typical conversion between these units, since available heat content
+        depends on the degree of U235 enrichement, the type of reactor, and whether the
+        fuel is just Uranium, or a mix of Uranium and Plutonium from decommissioned
+        nuclear weapons. See:
+
+        https://world-nuclear.org/information-library/facts-and-figures/heat-values-of-various-fuels.aspx
+
+        """
+        df = df.copy()
+
+        FuelFix = namedtuple("FuelFix", "fuel from_unit to_unit mult")
+        fuel_fixes = [
+            # US average coal heat content is 19.85 mmbtu/short ton
+            FuelFix("coal", "mmbtu", "ton", (1.0 / 19.85)),
+            FuelFix("coal", "btu", "ton", (1.0 / 19.85e6)),
+            # 2000 lbs per short ton
+            FuelFix("coal", "lbs", "ton", (1.0 / 2000.0)),
+            FuelFix("coal", "klbs", "ton", (1.0 / 2.0)),
+            # 42 gallons per barrel. Seriously, who makes up these units?
+            FuelFix("oil", "gal", "bbl", (1.0 / 42.0)),
+            FuelFix("oil", "kgal", "bbl", (1000.0 / 42.0)),
+            # On average a "ton of oil equivalent" is 7.46 barrels
+            FuelFix("oil", "ton", "bbl", 7.46),
+            FuelFix("gas", "mmbtu", "mcf", (1.0 / 1.037)),
+            # Nuclear plants report either heat content or mass of heavy metal
+            # MW*days thermal to MWh thermal
+            FuelFix("nuclear", "mwdth", "mwhth", 24.0),
+            # Straight energy equivalence between BTU and MWh here:
+            FuelFix("nuclear", "mmmbtu", "mwhth", (1.0 / 3.412142)),
+            FuelFix("nuclear", "btu", "mwhth", (1.0 / 3412142)),
+            # Unclear if it's possible to convert heavy metal to heat reliably
+            FuelFix("nuclear", "grams", "kg", (1.0 / 1000)),
+        ]
+        for fix in fuel_fixes:
+            fuel_mask = df.fuel_type_code_pudl == fix.fuel
+            unit_mask = df.fuel_units == fix.from_unit
+            df.loc[(fuel_mask & unit_mask), "fuel_consumed_units"] *= fix.mult
+            df.loc[(fuel_mask & unit_mask), "fuel_cost_per_unit_burned"] /= fix.mult
+            df.loc[(fuel_mask & unit_mask), "fuel_cost_per_unit_delivered"] /= fix.mult
```

Comment:
the units are `*`ed while the `cost_per_unit`'s are `/`ed bc cost/unit. i think a little bit of inline comments in this section would be helpful bc it is a little dense.

Reply:
True, that's confusing. Will comment.